test/Request/SerializerTest.php

```diff
@@ -247,6 +247,31 @@ public function testCanDeserializeResponseWithHeaderContinuations($text) : void
         $this->assertSame('Baz; Bat', $request->getHeaderLine('X-Foo-Bar'));
     }
 
+    /** @return non-empty-array<non-empty-string, array{non-empty-string}> */
+    public function headersWithWhitespace()
+    {
+        return [
+            'no' => ["POST /foo HTTP/1.0\r\nContent-Type: text/plain\r\nX-Foo-Bar:Baz\r\n\r\nContent!"],
+            'leading' => ["POST /foo HTTP/1.0\r\nContent-Type: text/plain\r\nX-Foo-Bar: Baz\r\n\r\nContent!"],
+            'trailing' => ["POST /foo HTTP/1.0\r\nContent-Type: text/plain\r\nX-Foo-Bar:Baz \r\n\r\nContent!"],
+            'both' => ["POST /foo HTTP/1.0\r\nContent-Type: text/plain\r\nX-Foo-Bar: Baz \r\n\r\nContent!"],
+        ];
+    }
+
+    /**
+     * @dataProvider headersWithWhitespace
+     */
+    public function testDeserializationRemovesWhitespaceAroundValues(string $text): void
+    {
+        $request = Serializer::fromString($text);
+
+        $this->assertInstanceOf(RequestInterface::class, $request);
```

Comment:
Redundant

Reply:
Stolen from `testCanDeserializeResponseWithHeaderContinuations` (which appears to be misnamed), removed. Will open another PR to fix the method name of that one.